File: app/src/main/java/com/anokmik/tripassistant/model/ObservableTrip.java
package com.anokmik.tripassistant.model;

import android.databinding.BaseObservable;

import com.anokmik.persistence.model.Trip;

public final class ObservableTrip extends BaseObservable {

    private Trip trip;

    public ObservableTrip(Trip trip) {
        this.trip = trip;
    }

    public Trip get() {
        return trip;
    }

    public void set(Trip trip) {
        this.trip = trip;
        notifyChange();
    }

    public String getTitle() {
        return trip.title;
    }

    public void setTitle(String title) {
        trip.title = title;
    }

    public String getDescription() {
        return trip.description;
    }

    public void setDescription(String description) {
        trip.description = description;
    }

    public long getStartDate() {
        return trip.startDate;
    }

    public void setStartDate(long startDate) {
        trip.startDate = startDate;
        notifyChange();
    }

    public long getFinishDate() {
        return trip.finishDate;
    }

    public void setFinishDate(long finishDate) {
        trip.finishDate = finishDate;
        notifyChange();
    }

    public void save() {
        trip.save();
    }

    public void delete() {
        trip.delete();
    }

}

File: app/src/main/kotlin/com/anokmik/tripassistant/model/ObservableTrip.kt
package com.anokmik.tripassistant.model

import android.databinding.BaseObservable

import com.anokmik.persistence.model.Trip

class ObservableTrip(private var trip: Trip?) : BaseObservable() {

    fun get(): Trip? {
        return trip
    }

    fun set(trip: Trip?) {
        this.trip = trip
        notifyChange()
    }

    var title: String?
        get() = trip?.title
        set(title) {
            trip?.title = title
        }

    var description: String?
        get() = trip?.description
        set(description) {
            trip?.description = description
        }

    var startDate: Long?
        get() = trip?.startDate
        set(startDate) {
            trip?.startDate = startDate
            notifyChange()
        }

    var finishDate: Long?
        get() = trip?.finishDate
        set(finishDate) {
            trip?.finishDate = finishDate
            notifyChange()
        }

    fun save() = trip?.save()

    fun delete() = trip?.delete()

}